charts/streams-app/templates/_helpers.tpl

```diff
@@ -27,3 +27,31 @@ Create chart name and version as used by the chart label.
 {{- define "streams-app.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Define default annotations from .Values.annotations.
+This will be used across resources.
+*/}}
+{{- define "streams-app.annotations" -}}
+{{- if or .Values.annotations }}
+  annotations:
+{{- range $key, $value := .Values.annotations }}
+    {{ $key | quote }}: {{ $value | quote }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Define annotations helper for Deployment.
+Includes default annotations and conditionally adds consumerGroup.
+Only includes annotations if there is content.
+*/}}
+{{- define "streams-app.deployment-annotations" -}}
+{{- include "streams-app.annotations" . }}
```

Comment:
There will be an error if you have a consumerGroup but no other annotation, I think

Reply:
Should be fixed:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: bakdata-streams-app
  annotations:
    consumerGroup: "foo"
```